chisel/src/result.rs

```diff
@@ -0,0 +1,211 @@
+use std::error::Error;
+use std::fmt::{self, Display};
+use std::fs::write;
+use std::path::PathBuf;
+
+use crate::fail;
+
+use libchisel::ModuleError;
+
+use ansi_term::Colour::{Green, Red, Yellow};
+use hex::encode;
+use parity_wasm::elements::{serialize, serialize_to_file, Module};
+
+#[derive(Clone)]
+/// Main result structure returned by ChiselDriver, containing a manifest of modules executed and
+/// exposing methods to write output from translators and creators.
+pub struct ChiselResult(Vec<RSResult>);
+
+#[derive(Clone)]
+/// Ruleset result structure.
+pub struct RSResult {
```

Comment:
Why not `RulesetResult`?

Reply:
Looks kinda like I spammed every letter in the word "Result" and then managed to finally type it correctly.